File: BinarySearch.py

def binary_Search(num_list, num):
  var mid_val;
  low_val = 0;
  high_val = len(num_list) - 1;
  result = False;
  while (high_val >= low_val not result):
        mid_val = (high_val + low_val)/2;
        if (num_list[mid_val] == num :
        result = True;
        else:
            if(num < num_list[mid_val]:
            high_val = mid_val - 1;
            else:
                low_val = mid_val + 1;
  return result
  
  print(binary_Search([1,3,5,7,9], 4)) //Output = False
  print(binary_Search([2,4,6,8,10], 6)) // Output = True
           
  

File: BinarySearch.cpp
//
//  main.cpp
//  BinarySearch
//
//  Created by Nkechi Iregbulem on 2/27/19.
//  Copyright © 2019 Nkechi Iregbulem. All rights reserved.
//

#include <iostream>
#include <string>
using namespace std;
#include <algorithm>
#include <functional>

// Binary Search Algorithm to return the value of key at given index of an array
int BinarySearch( int array[], int arraySize, int key) {
    int mid_val, low_val = 0, high_val = arraySize - 1;
    
    while (high_val >= low_val) {
        mid_val = (high_val + low_val)/2;
        if (array[mid_val] < key) {
            low_val = mid_val + 1;
        }
        else if(array[mid_val] > key) {
            high_val = mid_val - 1;
        }
        else {
            return mid_val;
        }
        return -1; // if not found
    }
    
}
    


int main() {
    
   const int array[] = {12, 24, 36, 48, 60, 72, 84, 96, 108 };
    const int arraySize = 9;
    int i, key, keyIndex;
    cout << "Enter a value " << endl; cin >> key;
    keyIndex = BinarySearch(array, arraySize, key);
    if (keyIndex == -1 ) {
        cout << key << " was not found" << endl;
    }
    else {
        cout << key << " at index " << keyIndex << "." << endl;
    }
    
    

    
    
    return 0;
}
